elements/pf-button/pf-button.ts

```diff
@@ -182,22 +185,71 @@ export class PfButton extends BaseButton {
    */
   @property({ reflect: true }) variant: ButtonVariant = 'primary';
 
-  protected override get hasIcon() {
-    return !!this.icon || !!this.loading;
-  }
+  /** Not as urgent as danger */
+  @property({ type: Boolean, reflect: true }) warning = false;
+
+  /**
+   * Use danger buttons for actions a user can take that are potentially
+   * destructive or difficult/impossible to undo, like deleting or removing
+   * user data.
+   */
+  @property({ type: Boolean, reflect: true }) danger = false;
+
+  /** Applies plain styles */
+  @property({ type: Boolean, reflect: true }) plain = false;
+
+  /** Changes the size of the button. */
+  @property({ reflect: true }) size?: 'small' | 'large';
+
+  /** Icon set for the `icon` property */
+  @property({ attribute: 'icon-set' }) iconSet?: string;
 
-  protected override renderDefaultIcon() {
+  /** Shorthand for the `icon` slot, the value is icon name */
+  @property() icon?: string;
+
+  /** Represents the state of a stateful button */
+  @property({ type: Boolean, reflect: true }) loading = false;
+
+  #internals = new InternalsController(this);
+
+  override render() {
+    const hasIcon = !!this.icon || !!this.loading;
     return html`
-      <pf-icon
-          icon="${ifDefined(this.icon)}"
-          set="${ifDefined(this.iconSet)}"
-          ?hidden="${!this.icon}"></pf-icon>
-      <pf-spinner
-          ?hidden="${!this.loading}"
-          size="md"
-          aria-label="${this.getAttribute('loading-label') ?? 'loading'}"></pf-spinner>
+      <button aria-label="${ifDefined(this.label)}"
+              class="${classMap({ hasIcon })}"
+              part="button"
+              type="${ifDefined(this.type)}"
+              value="${ifDefined(this.value)}"
+              @click="${this.#onClick}"
+              ?disabled="${this.disabled || this.#internals.formDisabled}">
+        <slot id="icon" part="icon" aria-hidden="true" name="icon">
+          <pf-icon
+              icon="${ifDefined(this.icon)}"
+              set="${ifDefined(this.iconSet)}"
+              ?hidden="${!this.icon}"></pf-icon>
+          <pf-spinner
+              ?hidden="${!this.loading}"
+              size="md"
+              aria-label="${this.getAttribute('loading-label') ?? 'loading'}"></pf-spinner>
```

Comment:
Not related to this PR specifically but while I'm reviewing...   

If a icon is given along with the loading being set 
```
<pf-button icon="link" id="loading-logs">Resume loading logs</pf-button>
```
You end up with stacked icon and spinner.  Should `pfe-icon` be hidden if loading?  I couldn't find anything specific in PF that mentioned this combo.

```Suggestion
          <pf-icon
              icon="${ifDefined(this.icon)}"
              set="${ifDefined(this.iconSet)}"
              ?hidden="${!this.icon} || ${this.loading}"></pf-icon>
          <pf-spinner
              ?hidden="${!this.loading}"
              size="md"
              aria-label="${this.getAttribute('loading-label') ?? 'loading'}"></pf-spinner>
```

Reply:
sounds reasonable